### PR TITLE
[#63] Fixes kafka reporter.

### PR DESCRIPTION
**Pull request by jcchavezs**
This PR fixes the `kafka` reporter sending array of objects as messages instead of a single object.

Ping @basvanbeek

### CODE REVIEW

File: reporter/kafka/kafka.go [hunk 1]
```diff
@@ -54,47 +54,49 @@ func Topic(t string) ReporterOption {
...
+	for pe := range r.producer.Errors() {
+		r.logger.Print("msg", pe.Msg, "err", pe.Err, "result", "failed to produce msg")
 	}
 }
 
-func (c *kafkaReporter) Send(s model.SpanModel) {
-	m, err := json.Marshal(s)
+func (r *kafkaReporter) Send(s model.SpanModel) {
+	// Zipkin expects the message to be wrapped in an array
+	ss := []model.SpanModel{s}
```
Comment (jcchavezs) [inline reporter/kafka/kafka.go]: This is the change that wraps the span with an array.

File: reporter/kafka/kafka_test.go [hunk 2]
```diff
@@ -116,90 +117,98 @@ func TestKafkaErrors(t *testing.T) {
...
 	for _, want := range spans {
-		_ = sendSpan(t, c, p, *want)
+		message := sendSpan(t, c, p, *want)
+		messageBody, err := message.Value.Encode()
+		if err != nil {
+			t.Errorf("unexpected error: %s", err.Error())
+		}
+
+		json.Unmarshal(messageBody, &have)
+		testEqual(t, want, &have[0])
```
Comment (jcchavezs) [inline reporter/kafka/kafka_test.go]: This test checks the expected and the actual message.

File: reporter/kafka/kafka_test.go [hunk 3]
```diff
@@ -116,90 +117,98 @@ func TestKafkaErrors(t *testing.T) {
...
 		}
 	}()
 
-	c.Send(s)
+	r.Send(s)
 
-	if !<-rcvd {
-		t.Fatal("span message was not produced")
+	if !<-received {
+		t.Fatal("expected span message produced")
```
Comment (basvanbeek) [inline reporter/kafka/kafka_test.go]: please reword this fatal message... this is a timeout we didn't expect right?

File: reporter/kafka/kafka.go [hunk 4]
```diff
@@ -54,47 +54,49 @@ func Topic(t string) ReporterOption {
 // NewReporter returns a new Kafka-backed Reporter. address should be a slice of
 // TCP endpoints of the form "host:port".
 func NewReporter(address []string, options ...ReporterOption) (reporter.Reporter, error) {
-	c := &kafkaReporter{
+	r := &kafkaReporter{
```
Comment (basvanbeek) [inline reporter/kafka/kafka.go]: yes... better 👍

**Pull request merged**